code/game/machinery/kitchen/microwave.dm

```diff
@@ -435,16 +435,20 @@
 	return ffuu
 
 /obj/machinery/microwave/CtrlClick(mob/user)
-    if(!user.incapacitated() && Adjacent(user) && user.dexterity_check() && anchored)
-        cook() //Cook checks for power, brokenness, and contents internally
-        return
-    return ..()
+	if(!user.incapacitated() && Adjacent(user) && user.dexterity_check() && anchored)
+		if(issilicon(user) && !attack_ai(user))
```

Comment:
It'd probably be best to just make a list somewhere of the modules allowed to access the machine, and check if the borg's module is one of the ones in the list when it attempts to do so.